pkg/datapath/iptables/iptables.go

```diff
@@ -493,33 +355,40 @@ func (m *IptablesManager) SupportsOriginalSourceAddr() bool {
 }
 
 // removeRules removes iptables rules installed by Cilium.
-func (m *IptablesManager) removeRules(prefix string, quiet bool) {
+func (m *IptablesManager) removeRules(prefix string) error {
 	// Set of tables that have had iptables rules in any Cilium version
 	tables := []string{"nat", "mangle", "raw", "filter"}
 	for _, t := range tables {
-		m.removeCiliumRules(t, ip4tables, prefix+"CILIUM_")
-	}
+		if err := m.removeCiliumRules(t, ip4tables, prefix+"CILIUM_"); err != nil {
+			return err
+		}
 
-	// Set of tables that have had ip6tables rules in any Cilium version
-	if m.haveIp6tables {
-		tables6 := []string{"nat", "mangle", "raw", "filter"}
-		for _, t := range tables6 {
-			m.removeCiliumRules(t, ip6tables, prefix+"CILIUM_")
+		if m.haveIp6tables {
+			if err := m.removeCiliumRules(t, ip6tables, prefix+"CILIUM_"); err != nil {
+				return err
+			}
 		}
 	}
 
 	for _, c := range ciliumChains {
 		c.name = prefix + c.name
-		c.remove(quiet)
+		if err := c.remove(true, m.haveIp6tables); err != nil {
```

Comment:
This also changes behaviour for IPv4, no? Old code's behaviour was effectively:

```suggestion
		if err := c.remove(option.Config.EnableIPv4, m.haveIp6tables); err != nil {
```

Not saying that the new code is wrong (it matches what we do for IPv6 now). But the commit message makes it sound like we're only changing IPv6 ...

Reply:
Yes, I think you are right :+1: The idea was to always try to remove/rename all old rules, regardless of what IP family is enabled (the commit mentions only IPv6 as probably that was what initially prompted me to fix this)